From c1ea43a45f4588f21752b0ad679c43350a9c8905 Mon Sep 17 00:00:00 2001
Message-ID: <c1ea43a45f4588f21752b0ad679c43350a9c8905.1698604357.git.vivien@planete-kraus.eu>
From: Vivien Kraus <vivien@planete-kraus.eu>
Date: Sun, 29 Oct 2023 19:00:44 +0100
Subject: [PATCH] Install gmobile as a shared library.

Tracked at https://gitlab.gnome.org/guidog/gmobile/-/issues/1

* src/meson.build: Install the header files.  Import pkgconfig.
Generate a pkg-config definition.
(gm_lib): use "library", not "static_library".  Install it.
(gmobile_gir): Install it.
* meson.build: Install gm-config.h.
---
 meson.build     |  1 +
 src/meson.build | 22 +++++++++++++++++++---
 2 files changed, 20 insertions(+), 3 deletions(-)

diff --git a/meson.build b/meson.build
index e9f6c62..51ebeac 100644
--- a/meson.build
+++ b/meson.build
@@ -83,6 +83,7 @@ root_inc = include_directories('.')
 configure_file(
   output: 'gm-config.h',
   configuration: config_h,
+  install_dir: get_option('includedir')
 )
 
 subdir('data')
diff --git a/src/meson.build b/src/meson.build
index ee98a39..3dedbe4 100644
--- a/src/meson.build
+++ b/src/meson.build
@@ -1,3 +1,5 @@
+pkg = import('pkgconfig')
+
 gm_deps = [
   gio_dep,
   glib_dep,
@@ -37,18 +39,33 @@ gm_c_args = [
   '-DG_LOG_DOMAIN="gmobile"',
 ]
 
-gm_lib = static_library(
+gm_lib = library(
   'gmobile',
   gm_sources,
   include_directories: root_inc,
   c_args: gm_c_args,
-  dependencies: gm_deps)
+  dependencies: gm_deps,
+  install: true)
+
+pkg.generate(gm_lib)
 
 gmobile_dep = declare_dependency(
   include_directories: [root_inc, include_directories('.')],
   dependencies: gm_deps,
   link_with: gm_lib)
 
+install_headers(
+  'gmobile.h',
+  'gm-cutout.h',
+  'gm-device-info.h',
+  'gm-device-tree.h',
+  'gm-display-panel.h',
+  'gm-error.h',
+  'gm-main.h',
+  'gm-rect.h',
+  'gm-svg-path.h',
+  'gm-timeout.h')
+
 if get_option('gtk_doc')
   gmobile_gir_extra_args = [
     '--c-include=gmobile.h',
@@ -66,7 +83,6 @@ if get_option('gtk_doc')
        identifier_prefix: 'Gm',
                link_with: gm_lib,
                 includes: ['Gio-2.0'],
-                 install: false,
               extra_args: gmobile_gir_extra_args,
    )
 endif

base-commit: 382fc89472176d2f1d435517cad53d969d8b8eff
-- 
2.41.0

